test/es6/classes.js

```diff
@@ -1021,7 +1021,23 @@ var tests = [
             for (s of new ClassWithArgumentsAndCallerComputedNameGeneratorMethods().caller()) {}
             assert.areEqual('456', s, "s of new ClassWithArgumentsAndCallerComputedNameGeneratorMethods().caller() === '456'");
         }
-    }
+    },
+    {
+        name: "toString on constructor should return class declaration or expression",
+        body: function () {
+            var B = class { };
+            var A = class A extends B { constructor (...args) { super(...args); }  set x(a) { this._x = a; } set y(a) { this._y = a; } };
+            var C = class C { set x(a){ this._x = a; } set y(a){ this._y = a; }};
+            var D = class D { constructor() {}  get x() { return 0; } };
+            var E = D;
+
+            assert.areEqual("class A extends B { constructor (...args) { super(...args); }  set x(a) { this._x = a; } set y(a) { this._y = a; } }", A.prototype.constructor.toString());
+            assert.areEqual("class { }", B.prototype.constructor.toString());
+            assert.areEqual("class C { set x(a){ this._x = a; } set y(a){ this._y = a; }}", C.prototype.constructor.toString());
+            assert.areEqual("class D { constructor() {}  get x() { return 0; } }", D.prototype.constructor.toString());
+            assert.areEqual("class D { constructor() {}  get x() { return 0; } }", E.prototype.constructor.toString());

```

Comment:
All these classes are expressions, we should also test declarations.  Also I'm curious about definitions that contain newlines.  I imagine that since it uses the same mechanisms as normal function tostring that newlines should be fine, but one test would be good to have for comprehensiveness.